### PR TITLE
@astrojs/image "base" astro.config.mjs path configuration doesn´t works on SSR build and preview

### LINKED ISSUE #5807: @astro/image subpath configuration doesn´t works on SSR with sharp or squoosh endpoints
### What version of `astro` are you using?

1.9.1

### Are you using an SSR adapter? If so, which one?

Yes / NodeJS

### What package manager are you using?

yarn

### What operating system are you using?

Window or Ubuntu 22.04

### Describe the Bug

When SSR mode enabled, subpath "base"configuration on astro.config.mjs are ignored after building the project. Works fine on dev mode, but doesn´t includes the subpath after build. Can be reproduced in preview mode.

Would like to submit a PR, like everyone time is short, but will do my best.

### Link to Minimal Reproducible Example

https://github.com/dsclearuc/astroimage_ssr_problem_image

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by dsclearuc**
@atrojs/image - SSR build && preview problem with base config settings on astro.config.mjs

Fix to when astro base path config enabled, SSR build and preview not working.
Fixes #5807

## Changes

- Includes base path to GetImage exported function
- Updates Astro file filter to consider sub paths followed by "/_image/.." pattern

## Testing

Not updated tests. Needs technical help for this step;

## Docs
No documentation required. Bugfix. The module should work with base configuration without further configs or additional code.

### CODE REVIEW

File: packages/integrations/image/src/lib/get-image.ts [hunk 1]
```diff
@@ -149,7 +150,9 @@ export async function getImage(
 		src = `${imgSrc}?${searchParams.toString()}`;
 	} else {
 		searchParams.set('href', imgSrc);
-		src = `/_image?${searchParams.toString()}`;
+		// Takes account astro base path configuration
+		// @ts-ignore
+		src = `${import.meta.env?.BASE_URL ? '/' + removeTrailingForwardSlash(import.meta.env.BASE_URL) : ''}/_image?${searchParams.toString()}`;
```
Comment (rishi-raj-jain) [inline packages/integrations/image/src/lib/get-image.ts:155]: How about appendForwardTrailingSlash(import.meta.env.BASE_URL)_image?${searchParams.toString()} instead?

**Review by dsclearuc — COMMENTED** [review]
@rishi-raj-jain You're right. Tks

**Comment by matthewp** [comment]
Looks like this breaks a few things :( @dsclearuc are you still planning on working on it?

**Comment by rishi-raj-jain** [comment]
What does this break though? I can pick it up.

**Comment by matthewp** [comment]
A few image tests, see the CI results: https://github.com/withastro/astro/actions/runs/4128019902/jobs/7131989900

**Comment by dsclearuc** [comment]
@rishi-raj-jain Tks for the support, on a rally here in my work, not even seeing emails :-(. @matthewp I think will be necessary some CI tests for the subpath configuration.

**Comment by matthewp** [comment]
No problem @dsclearuc. I'm going to close as this is stale. If you get the time to work on it again, please submit a new PR, thanks!

**Pull request closed without merging**